### PR TITLE
Check cluster UUID on watch calls.

### PR DESCRIPTION
**Pull request by fasaxc**
Cory, I think this should do the trick for the basic check that we discussed.  I didn't get it packaged today (got sidetracked by more pressing issues that were blocking the bobcats).

### CODE REVIEW

File: src/etcd/client.py [hunk 1]
```diff
@@ -572,11 +575,29 @@ def api_execute(self, path, method, params=None, timeout=None):
...
+                        self._cluster_uuid and
+                        (cluster_id != self._cluster_uuid)):
+                    raise etcd.EtcdClusterIdChanged(
+                        'The UUID of the cluster changed from {} to '
+                        '{}.'.format(self._cluster_uuid, cluster_id))
+                elif not self._cluster_uuid:
+                    self._cluster_uuid = cluster_id
+
+                # Force synchronous load of data before we return.
+                _ = response.data
```
Comment (Lukasa) [inline src/etcd/client.py:604]: I assume that this is here to keep the same behaviour as the old `preload_content=True` logic, and that you'll replace it when you drop the next patch.

File: src/etcd/client.py [hunk 2]
```diff
@@ -572,11 +575,29 @@ def api_execute(self, path, method, params=None, timeout=None):
...
+                # below.
+                cluster_id = response.getheader("x-etcd-cluster-id")
+                if (check_cluster_uuid and
+                        self._cluster_uuid and
+                        (cluster_id != self._cluster_uuid)):
+                    raise etcd.EtcdClusterIdChanged(
+                        'The UUID of the cluster changed from {} to '
+                        '{}.'.format(self._cluster_uuid, cluster_id))
+                elif not self._cluster_uuid:
+                    self._cluster_uuid = cluster_id
```
Comment (Lukasa) [inline src/etcd/client.py:601]: I don't like having all this logic inside the `try`: can we pull it out, either into an `else` block or after the `try...except`?

**Comment by Lukasa** [comment]
Some small notes.

File: src/etcd/client.py (hunk 1 above)
Comment (fasaxc) [inline src/etcd/client.py:604]: Yeh, I was trying to be defensive and not rework the code paths too much in this patch.

**Pull request merged**